ivcfmerge/batch.py

```diff
@@ -0,0 +1,36 @@
+from contextlib import ExitStack
+from queue import Queue
+from tempfile import NamedTemporaryFile
+
+from . import ivcfmerge
+
+
+def ivcfmerge_batch(input_paths, output_path, batch_size=1000):
+    if batch_size < 2 or batch_size > len(input_paths):
+        batch_size = len(input_paths)
+
+    if batch_size == len(input_paths):
+        batch = input_paths
+    else:
+        work_queue = Queue()
+        [work_queue.put(item) for item in input_paths]
+
+        while True:
+            batch = []
+            while not work_queue.empty() and len(batch) < batch_size:
+                batch.append(work_queue.get_nowait())
+
+            if len(batch) < batch_size or work_queue.empty():
+                break
+
+            tmp_filename = NamedTemporaryFile().name
+            _ivcfmerge(batch, tmp_filename)
+            work_queue.put(tmp_filename)
```

Comment:
I have another thought on the ordering of samples in the merged VCFs. I think it is reasonable for a user to expect that the order will remain the same as they appear in the input. This requirement is not listed in the ticket. I will add another ticket to meet this requirement.